Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -180,17 +186,30 @@ def setup(self, dcfg=None, pstate=None, initialdlstatus=None, lm_network_engine_
 
                 self._logger.debug(u"setup: initialdlstatus %s %s", hexlify(self.tdef.get_infohash()), initialdlstatus)
 
-                self.create_engine_wrapper(lm_network_engine_wrapper_created_callback, pstate,
-                                           initialdlstatus=initialdlstatus, wrapperDelay=wrapperDelay)
+                def schedule_create_engine():
+                    if self.can_create_engine_wrapper():
+                        # Subroutine that can be scheduled by the taskManager as it cannot handle deferreds.
+                        def create_engine():
+                            self.cew_scheduled = True
+                            create_engine_wrapper_deferred = self.network_create_engine_wrapper(pstate, initialdlstatus)
+                            create_engine_wrapper_deferred.chainDeferred(deferred)
+
+                        # Schedule the call to create the engine with a delay defined by wrapperDelay
+                        self.session.lm.threadpool.add_task(create_engine, wrapperDelay)
+                    else:
+                        self.session.lm.threadpool.add_task(schedule_create_engine, 5)
+
+                schedule_create_engine()
```

Comment:
Oh, sorry I was in a hurry and didn't see line 200.

I guess it would be prettier if you used a LoopingCall that stops itself when can_create_engine_wrapper() is True...